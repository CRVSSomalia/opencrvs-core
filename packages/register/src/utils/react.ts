export type ComponentProps<C> = C extends React.Component<infer P, any>
  ? P
  : never
