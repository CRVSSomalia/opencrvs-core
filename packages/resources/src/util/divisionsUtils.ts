import { ILocation } from 'src/features/administrative/service'

export const appendDivisions = (
  divisions: ILocation[],
  newDivisions: ILocation[]
): ILocation[] => {
  return newDivisions.reduce((total, item) => {
    total.push(item)
    return total
  }, divisions)
}
