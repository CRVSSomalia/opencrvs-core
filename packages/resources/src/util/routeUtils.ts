export const replaceParamsInRoute = (
  route: string,
  params: string[]
): string => {
  return route.replace(/%(\d+)/g, (_, n) => params[+n - 1])
}
