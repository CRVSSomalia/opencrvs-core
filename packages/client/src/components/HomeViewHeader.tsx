/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
 * graphic logo are (registered/a) trademark(s) of Plan International.
 */
import * as React from 'react'
import { IViewHeadingProps, ViewHeading } from '@client/components/ViewHeading'
import { Header } from '@opencrvs/components/lib/interface'
import { ActionList } from '@opencrvs/components/lib/buttons'
import styled from '@client/styledComponents'
import { TopMenu } from '@client/components/TopMenu'
import { Logo } from '@opencrvs/components/lib/icons'
import ConnectivityStatus from '@client/components/ConnectivityStatus'
import { BodyContent } from '@opencrvs/components/lib/layout'

const HeaderWrapper = styled(Header)`
  display: block;
  justify-content: flex-end;
  padding-bottom: 50px;
  min-height: 280px;

  /* stylelint-disable */
  + ${ActionList} {
    /* stylelint-enable */
    margin-top: -80px;
  }
`

const StyledLogo = styled(Logo)`
  margin: -71px auto auto 20px;
`

export class HomeViewHeader extends React.Component<IViewHeadingProps> {
  render() {
    const { title, description, id } = this.props
    return (
      <HeaderWrapper>
        <BodyContent>
          <ConnectivityStatus />
          <TopMenu hideBackButton={true} />
          <StyledLogo />
          <ViewHeading {...{ title, description, id }} />
          {this.props.children}
        </BodyContent>
      </HeaderWrapper>
    )
  }
}
