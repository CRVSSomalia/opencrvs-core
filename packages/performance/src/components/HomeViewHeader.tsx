/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
 * graphic logo are (registered/a) trademark(s) of Plan International.
 */
import * as React from 'react'
import {
  IViewHeadingProps,
  ViewHeading
} from '@performance/components/ViewHeading'
import { Header } from '@opencrvs/components/lib/interface'
import { ActionList } from '@opencrvs/components/lib/buttons'
import styled from '@performance/styledComponents'
import { TopMenu } from '@performance/components/TopMenu'
import { Logo, Offline, Online } from '@opencrvs/components/lib/icons'

const StretchedHeader = styled(Header)`
  justify-content: flex-end;
  padding-bottom: 50px;
  min-height: 280px;

  /* stylelint-disable */
  + ${ActionList} {
    /* stylelint-enable */
    margin-top: -80px;
  }
`

const StyledLogo = styled(Logo)`
  margin: -71px auto auto 17px;
`

const StyledOnline = styled.div`
  position: absolute;
  top: 32px;
  right: 234px;
`

const StyledOffline = styled.div`
  position: absolute;
  top: 32px;
  right: 234px;
`

export class HomeViewHeader extends React.Component<IViewHeadingProps> {
  render() {
    const { title, description, id } = this.props
    return (
      <StretchedHeader {...this.props}>
        {navigator.onLine ? (
          <StyledOnline>
            <Online />
          </StyledOnline>
        ) : (
          <StyledOffline>
            <Offline />
          </StyledOffline>
        )}
        <TopMenu hideBackButton={true} />
        <StyledLogo />
        <ViewHeading {...{ title, description, id }} />
        {this.props.children}
      </StretchedHeader>
    )
  }
}
