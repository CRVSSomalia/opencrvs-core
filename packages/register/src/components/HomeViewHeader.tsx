import * as React from 'react'

import { Header } from '@opencrvs/components/lib/interface'
import { ActionList } from '@opencrvs/components/lib/buttons'
import styled from '../styled-components'
import { TopMenu } from '../components/TopMenu'
import Logo from './Logo'

const StretchedHeader = styled(Header)`
  justify-content: flex-end;
  padding-bottom: 50px;
  min-height: 280px;

  /* stylelint-disable */
  + ${ActionList} {
    /* stylelint-enable */
    margin-top: -80px;
  }
`

const StyledLogo = styled(Logo)`
  margin: -40px auto auto 40px;
`

export class HomeViewHeader extends React.Component {
  render() {
    return (
      <StretchedHeader {...this.props}>
        <TopMenu hideBackButton={true} />
        <StyledLogo />
        {this.props.children}
      </StretchedHeader>
    )
  }
}
