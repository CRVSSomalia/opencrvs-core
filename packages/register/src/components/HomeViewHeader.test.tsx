import * as React from 'react'
import { createShallowRenderedComponent } from '../tests/util'
import { HomeViewHeader } from './HomeViewHeader'

describe('home view header component', () => {
  const testComponent = createShallowRenderedComponent(<HomeViewHeader />)

  it('renders without crashing', () => {
    expect(testComponent).toMatchSnapshot()
  })
})
