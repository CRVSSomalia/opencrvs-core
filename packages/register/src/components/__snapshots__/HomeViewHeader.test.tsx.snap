// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home view header component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeViewHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Connect(InjectIntl(TopMenuComponent))
          hideBackButton={true}
        />,
        <Styled(Logo) />,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "hideBackButton": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Connect(InjectIntl(TopMenuComponent))
            hideBackButton={true}
          />,
          <Styled(Logo) />,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "hideBackButton": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
